In [1]:
import pandas as pd

C:\Users\Imran\anaconda3\Lib\site-packages\pandas\core\arrays\masked.py:60: UserWarning: Pandas requires version '1.3.6' or newer of 'bottleneck' (version '1.3.5' currently installed).
  from pandas.core import (


In [2]:
data = pd.read_excel("new.xlsx")

In [3]:
data.columns

Index(['Timestamp', 'Name', 'Age', 'Gender', 'Gmail', 'Field_of_interest',
       'Internet_usage_activity', 'Time_spent', 'Productivity_Status',
       'First_Internet_preferred_activity',
       'Second_Internet_preferred_activity',
       'Third_Internet_preferred_activity', 'Entertainment_time_on_internet',
       'Platform_used_for_entertainment', 'Usage_frequency',
       'Proficiency_in_programming/software', 'programming_language/software',
       'Learning_site', 'City/Province'],
      dtype='object')

In [4]:
data.Age.value_counts()

Age
16 - 24         25
Less than 16     6
More than 24     1
Name: count, dtype: int64

Majority of the respondents are between the age of 16-24

In [5]:
data.Gender.value_counts(normalize = True) * 100

Gender
Female    84.375
Male      15.625
Name: proportion, dtype: float64

84% of the respondents are females.

In [6]:
data

Timestamp                Name           Age  Gender  \
0  2024-08-04 21:49:51.823       Fatima Zafar        16 - 24  Female   
1  2024-08-06 10:17:40.980     Jaweria Nadeem        16 - 24  Female   
2  2024-08-06 23:05:41.786       Maryam Farhan       16 - 24  Female   
3  2024-08-06 23:07:04.166         Hafsa Rafiq       16 - 24  Female   
4  2024-08-06 23:19:36.507               Hibba       16 - 24  Female   
5  2024-08-06 23:41:19.690              Ayesha       16 - 24  Female   
6  2024-08-06 23:48:26.385  Lightning McQueen        16 - 24  Female   
7  2024-08-06 23:58:05.108        Zeenat Ilyas       16 - 24  Female   
8  2024-08-07 00:01:43.771        Umaima Zafar  Less than 16  Female   
9  2024-08-07 00:02:18.850           Samawiyah  Less than 16  Female   
10 2024-08-07 00:33:27.046      Maryam Faisal        16 - 24  Female   
11 2024-08-07 00:38:19.504           Shehzeel        16 - 24  Female   
12 2024-08-07 13:06:11.283             Anzalna       16 - 24  Female   
13 2024-08-07 14:42:07.768      Pariwash Imran       16 - 24  Female   
14 2024-08-09 16:27:38.512       Maliha Ajmal   Less than 16  Female   
15 2024-09-04 01:21:28.006       Humais Mujeeb       16 - 24    Male   
16 2024-09-04 14:36:44.186  Syeda meerab afzal       16 - 24  Female   
17 2024-09-04 14:42:04.176             Rimsha        16 - 24  Female   
18 2024-09-04 14:49:49.998      Rumaisa Waseem       16 - 24  Female   
19 2024-09-04 14:57:10.203        Nimra Tayyab       16 - 24  Female   
20 2024-09-04 15:43:38.289        Azalfa Zakir  Less than 16  Female   
21 2024-09-04 16:48:21.269   Abdul Rehman Khan       16 - 24    Male   
22 2024-09-04 16:49:03.913        Sufyan Zafar       16 - 24    Male   
23 2024-09-04 17:23:23.204               Aaiza       16 - 24  Female   
24 2024-09-04 17:52:29.758      Muhammad Aamir  More than 24    Male   
25 2024-09-04 20:11:33.566     Khadija riffat   Less than 16  Female   
26 2024-09-05 00:07:07.093       Muhammad Ahad       16 - 24    Male   
27 2024-09-05 00:24:19.551        Ayesha Aqeel       16 - 24  Female   
28 2024-09-05 21:28:17.790                Eman       16 - 24  Female   
29 2024-09-06 00:34:32.238               Yumna  Less than 16  Female   
30 2024-09-06 14:11:15.613        Tooba Khanum       16 - 24  Female   
31 2024-09-08 12:16:38.255        Ayesha khan        16 - 24  Female   

                             Gmail     Field_of_interest  \
0         zafarfatima061@gmail.com           Visual Arts   
1        jawerianadeem41@gmail.com                 Jobs    
2        farhanmaryam656@gmail.com                    CS   
3              rhafsa408@gmail.com         Cybersecurity   
4          hibba.sep2006@gmail.com        Cybersecurity    
5        ayeshaarif01245@gmail.com                    CS   
6   aiman123.siddiqui123@gmail.com     Fashion designer    
7         zeenatilyas995@gmail.com             Medical     
8          umaimazafar62@gmail.com                    CS   
9        samaviashakeel2@gmail.com                Doctor   
10     www.inshaalfaisal@gmail.com                    CS   
11            affanalim@gmail.com               Medical    
12      anzalnasiddiqui7@gamil.com           Engineering   
13     pariwashimran1212@gmail.com              Medical    
14         malihaajmal26@gmail.com                    CS   
15           humaisworks@gmail.com        Media And Arts   
16       Meerabafzal5507@gmail.com                    CS   
17        rimshawaseem29@gmail.com           Visual Arts   
18           rumaisa.w05@gmail.com                    CS   
19        nimratayyab320@gmail.com                    CS   
20           azalfazakir@gmail.com              Medical    
21       kabdulrehman755@gmail.com       Web Development   
22        sufyanzafar500@gmail.com      Export Marketing   
23          Aaiza.adil@outlook.com        Data analysis    
24            aamirfrazy@gmail.com              Teaching   
25       khadijariffat38@gmail.com          Engineering    
26   muhammadahadans

In [7]:
data.Field_of_interest.value_counts()

Field_of_interest
CS                      9
Medical                 3
Visual Arts             2
Export Marketing        1
Video editing           1
Social Work             1
Chartered Acountancy    1
Software Developer      1
Engineering             1
Teaching                1
Data analysis           1
Media And Arts          1
Web Development         1
Jobs                    1
Engineering             1
Doctor                  1
Medical                 1
Fashion designer        1
Cybersecurity           1
Cybersecurity           1
Lecturar                1
Name: count, dtype: int64

Majority of the respondents either are persuing or want to persue computer science or related field.

In [8]:
data["Entertainment_time_on_internet"].value_counts(normalize = True)*100

Entertainment_time_on_internet
More than 3 hours    46.875
Less than 3 hours    43.750
3 hours               9.375
Name: proportion, dtype: float64

In [15]:
data["Time_spent"].value_counts(normalize = True)*100

Time_spent
Less than 3 hours    40.625
More than 3 hours    37.500
3 hours              21.875
Name: proportion, dtype: float64

Around 40.6% sample population spents less than 3 hours on internet.

In [10]:
data["Productivity_Status"].value_counts(normalize = True)*100

Productivity_Status
Yes    68.75
No     31.25
Name: proportion, dtype: float64

Majority of the people consider their activities on internet as productive.

In [16]:
data["Internet_usage_activity"].value_counts()

Internet_usage_activity
YouTube                   6
Studies                   6
Entertainment             5
Instagram                 2
Instagram                 2
Reading                   2
Studies                   1
studies                   1
Social Media              1
Ms Excel                  1
Entertainment             1
Facebook                  1
Information collection    1
Research                  1
Work                      1
Name: count, dtype: int64

Internet is most used by sample population for watching YouTube and studying.

In [12]:
data[data["Productivity_Status"] == "Yes"]["Internet_usage_activity"].value_counts()

Internet_usage_activity
Studies                   6
YouTube                   4
Entertainment             2
Studies                   1
studies                   1
Social Media              1
Ms Excel                  1
Entertainment             1
Facebook                  1
Information collection    1
Research                  1
Work                      1
Reading                   1
Name: count, dtype: int64

As expected, people who spent their time on internet for studies are more likely to consider it productive. However, significant population also consider entertainment as productive.

In [13]:
data["Platform_used_for_entertainment"].value_counts()

Platform_used_for_entertainment
YouTube                                          15
Instagram                                         5
Netflix, Instagram                                4
YouTube, Instagram                                2
Snapchat                                          1
Netflix                                           1
YouTube, Instagram, Google, Pinterest             1
YouTube, Snapchat, Instagram, Facebook            1
YouTube, TikTok                                   1
YouTube, WhatsApp, LinkedIn, GitHub, Facebook     1
Name: count, dtype: int64

In [14]:
data["Entertainment_time_on_internet"][data["Platform_used_for_entertainment"] == "YouTube"].value_counts(normalize=True)*100

Entertainment_time_on_internet
Less than 3 hours    53.333333
More than 3 hours    46.666667
Name: proportion, dtype: float64

47% of the population spents more than 3 hours on entertainment through internet.

In [19]:
data["First_Internet_preferred_activity"].value_counts()

First_Internet_preferred_activity
Studies           11
Entertainment      5
Work               4
Entertainment      4
Communication      4
Work               2
Study              1
Communication      1
Name: count, dtype: int64

For 11 out of 32 people, internet is most useful for learning.

In [21]:
data["Proficiency_in_programming/software"].value_counts()

Proficiency_in_programming/software
Yes    19
No     13
Name: count, dtype: int64

Majoirity of the people are proficient in software tools 

In [26]:
data["Field_of_interest"][data["Proficiency_in_programming/software"] == "Yes"].value_counts()

Field_of_interest
CS                      7
Visual Arts             1
Jobs                    1
Cybersecurity           1
Doctor                  1
Medical                 1
Media And Arts          1
Web Development         1
Data analysis           1
Engineering             1
Software Developer      1
Chartered Acountancy    1
Video editing           1
Name: count, dtype: int64

And, as expected, they are persuing computer science or computer related fields, except for one. 

In [36]:
data["Platform_used_for_entertainment"].value_counts()

Platform_used_for_entertainment
YouTube                                          15
Instagram                                         5
Netflix, Instagram                                4
YouTube, Instagram                                2
Snapchat                                          1
Netflix                                           1
YouTube, Instagram, Google, Pinterest             1
YouTube, Snapchat, Instagram, Facebook            1
YouTube, TikTok                                   1
YouTube, WhatsApp, LinkedIn, GitHub, Facebook     1
Name: count, dtype: int64

YouTube is the most popular productivity app.

In [46]:
data["programming_language/software"].str.split(",").str[0].value_counts()

programming_language/software
Python                                      4
C++                                         2
Html                                        2
Photoshop                                   1
Yes of course                               1
HTML                                        1
MS word                                     1
Html                                        1
Adobe Photoshop                             1
Adobe Creative Suite                        1
Graphic designing                           1
Backend developer Nodejs                    1
basic html                                  1
No                                          1
Illustrator photoshop lightroom premier     1
Name: count, dtype: int64

Majority of the population has expertise in python.